scripts/events_module/event_filters.py

```diff
@@ -0,0 +1,412 @@
+    factor = factor - round(
+        pow((clan_size / divider), 2)
+    )
+    if factor < 2 and game.clan.game_mode == "expanded":
+        factor = 2
+
+    trigger_value = round(factor * needed_amount, 2)
+
+    if "full" in trigger and needed_amount < clan_supply < trigger_value:
+        return True
+    if "excess" in trigger and clan_supply > trigger_value:
+        return True
+
+    # if it hasn't returned by now, it doesn't qualify
+    return False
+
+
+def event_for_herb_supply(trigger, supply_type, clan_size) -> bool:
+    """
+    checks if clan's herb supply qualifies for event
+    """
+    if "always" in trigger:
+        return True
+
+    herb_supply = game.clan.herbs.copy()
+    possible_herbs = HERBS
+    num_of_herbs = len(possible_herbs)
+
+    if not herb_supply and "low" in trigger:
+        return True
+
+    for herb in possible_herbs:
+        if herb not in herb_supply:
+            herb_supply[herb] = 0
+
+    needed_amount = clan_size * 2
+    half_amount = needed_amount / 2
+
+    if supply_type == "all_herb":
+        if "low" in trigger and len([x for x in herb_supply if herb_supply[x] < half_amount]) == num_of_herbs:
+            return True
+        elif "adequate" in trigger and len(
+                [x for x in herb_supply if half_amount < herb_supply[x] <= needed_amount]) == num_of_herbs:
+            return True
+        elif "full" in trigger and len(
+                [x for x in herb_supply if needed_amount < herb_supply[x] <= needed_amount * 2]) == num_of_herbs:
+            return True
+        elif "excess" in trigger and len(
+                [x for x in herb_supply if needed_amount * 2 < herb_supply[x]]) == num_of_herbs:
+            return True
+
+        return False
+
+    elif supply_type == "any_herb":
+        if "low" in trigger and [x for x in herb_supply if herb_supply[x] < half_amount]:
+            return True
+        elif "adequate" in trigger and [x for x in herb_supply if half_amount < herb_supply[x] <= needed_amount]:
+            return True
+        elif "full" in trigger and [x for x in herb_supply if needed_amount < herb_supply[x] <= needed_amount * 2]:
+            return True
+        elif "excess" in trigger and [x for x in herb_supply if needed_amount * 2 < herb_supply[x]]:
+            return True
+
+        return False
+
+    else:
+        chosen_herb = supply_type
+        if chosen_herb not in possible_herbs:
+            print(f"WARNING: possible typo in supply constraint: {chosen_herb}")
+            return False
+        if "low" in trigger and herb_supply[chosen_herb] < half_amount:
+            return True
+        elif "adequate" in trigger and half_amount < herb_supply[chosen_herb] <= needed_amount:
+            return True
+        elif "full" in trigger and needed_amount < herb_supply[chosen_herb] <= needed_amount * 2:
+            return True
+        elif "excess" in trigger and needed_amount * 2 < herb_supply[chosen_herb]:
+            return True
```

Comment:
This whole block doesn't need to be in an `else` statement (I'm not brave enough to remove the indentations in a suggestion so I leave that as an exercise for the reader ;) )